packages/tools/devtools/devtools-core/src/test/DataVisualizerGraph.test.ts

```diff
@@ -205,4 +206,29 @@ describe("DataVisualizerGraph unit tests", () => {
 		};
 		expect(childCellTree).to.deep.equal(expectedChildCellTree);
 	});
+
+	it("Unknown object in Container Data", async () => {
+		const unknownObject = {};
+
+		const visualizer = new DataVisualizerGraph(
+			{
+				unknownObject: unknownObject as IFluidLoadable,
+			},
+			defaultVisualizers,
+			defaultEditors,
+		);
+
+		const childUnknownObject = await visualizer.render("unknown-object");
+		expect(childUnknownObject).to.deep.equal(undefined);
+	});
+
+	it("Empty Container Data", async () => {
+		// Pass in the empty containerData to the visualizer.
+		const emptyRecord: Record<string, IFluidLoadable> = {};
+
+		const visualizer = new DataVisualizerGraph(emptyRecord, defaultVisualizers, defaultEditors);
+
+		const childEmptyRecord = await visualizer.render("empty-containerData");
```

Comment:
This only verifies that we can't find something to render for key "empty-containerData". We want to verify that rendering the root handles of the visualizer returns an empty record when the containerData we gave it was empty.

Reply:
I made the following changes for the `emptyRecord` unit test:
- Call `renderRootHandles()`
- Compare the output with the empty record of `expectedChildEmptyRecord`

One thing that I am wondering is whether or not setting the type as `Record<string, RootHandleNode>` is correct.

`renderRootHandles()` method first initializes the empty Record as `const result: Record<string, RootHandleNode> = {};` and would return this right away as the `rootDataEntries` is empty (and this is why I used `RootHandleNode` as the expected type). 

But since `RootHandleNode` can either be `FluidHandleNode | UnknownObjectNode`, I am unsure if I should set more of less-generic type.

Reply:
Honestly, you could probably just pass in `{}` (and ignore any linter errors). I don't think you'll have to specify a type in that case.